### PR TITLE
[Ci] add Ci control of available services

### PR DESCRIPTION
**Pull request by Dima73**
-thanks  guys openATV

### CODE REVIEW

**Comment by WanWizard** [comment]
What does this do? CI helper?

**Comment by Dima73** [comment]
Usually, the CI module decodes only one service.
This patch adds a check for the availability service for recording and when switching channels.

**Comment by littlesat** [comment]
But thanks for the work... let the rest of the team also evaluatie this before I merge... eg as we do not have offline decoding functional a sanity timer check on timers might be welcome... but I see this patch is just doing more that that...

**Comment by littlesat** [comment]
And as the only description is just thanks atv... I do known exactly known what I merge

**Comment by littlesat** [comment]
When it is only for ci sanity timer check I Will not merge this one... then better help us to get offline decoding (when box is in stand-by) running... which makes this merge request not required...

**Comment by Dima73** [comment]
This is only the first step.
offline decoding != ci control
I checked the work of this patch.
Everything works great.
Recording from the module + live tv/recording softcam service.

**Comment by littlesat** [comment]
I see there is still some development...
But what exactly is it doing? Can you please describe... Does it also do dual CIs..?

**Comment by Dima73** [comment]
Yes,simultaneous decoding of two services, if the module allows it,e.g. module "AlphaCrypt".

**Comment by eriksl** [comment]
This commit mixes whitespace changes and functionality.

**Comment by Dima73** [comment]
Where exactly?
I'll fix it.

File: lib/python/Plugins/SystemPlugins/CommonInterfaceAssignment/plugin.py [hunk 1]
```diff
@@ -161,7 +152,7 @@ def __init__(self, session, ci_slot="9"):
 		self.loadXML()
 		# if config mode !=advanced autoselect any caid
 		if config.usage.setup_level.index <= 1: # advanced
-			self.selectedcaid=self.caidlist
```
Comment (eriksl) [inline lib/python/Plugins/SystemPlugins/CommonInterfaceAssignment/plugin.py:164]: whitespace

File: lib/python/Plugins/SystemPlugins/CommonInterfaceAssignment/plugin.py [hunk 2]
```diff
@@ -286,16 +282,15 @@ def getValue(definitions, default):
...
 				read_slot = getValue(slot.findall("id"), False).encode("UTF-8")
-				print "ci " + read_slot
-				i=0
+				i = 0
 				for caid in slot.findall("caid"):
 					read_caid = caid.get("id").encode("UTF-8")
 					self.selectedcaid.append((str(read_caid),str(read_caid),i))
 					self.usingcaid.append(long(read_caid,16))
-					i+=1
+					i += 1
```
Comment (eriksl) [inline lib/python/Plugins/SystemPlugins/CommonInterfaceAssignment/plugin.py:293]: whitespace

File: lib/python/Plugins/SystemPlugins/CommonInterfaceAssignment/plugin.py [hunk 3]
```diff
@@ -343,8 +337,8 @@ class easyCIconfigMenu(CIconfigMenu):
 		</screen>"""
 
 	def __init__(self, session, ci_slot="9"):
-
-		ci=ci_slot
```
Comment (eriksl) [inline lib/python/Plugins/SystemPlugins/CommonInterfaceAssignment/plugin.py:347]: whitespace

**Comment by eriksl** [comment]
I don't mind for once, but please keep our git neat and always check for new features + changes agains whitespace and cosmetic -> always two commits.

**Comment by Dima73** [comment]
Ok, I get it.
Cosmetics = separate commit

**Comment by eriksl** [comment]
Thumbs up :+1: :-)

**Pull request merged**